RecoParticleFlow/PFClusterProducer/plugins/Basic2DGenericPFlowPositionCalc.h

```diff
@@ -8,16 +8,61 @@
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 #include "RecoParticleFlow/PFClusterProducer/interface/CaloRecHitResolutionProvider.h"
+#include <tuple>
 
 class Basic2DGenericPFlowPositionCalc : public PFCPositionCalculatorBase {
  public:
   Basic2DGenericPFlowPositionCalc(const edm::ParameterSet& conf) :
     PFCPositionCalculatorBase(conf),    
     _posCalcNCrystals(conf.getParameter<int>("posCalcNCrystals")),
-    _logWeightDenom(1./conf.getParameter<double>("logWeightDenominator")),
     _minAllowedNorm(conf.getParameter<double>("minAllowedNormalization"))
-
   {  
+
+    std::vector<int> detectorEnum;
+    std::vector<int> depths;
+    std::vector<double> logWeightDenom;
+    std::vector<float> logWeightDenomInv;
+
+    if(conf.exists("logWeightDenominatorByDetector")) {
+
+      const std::vector<edm::ParameterSet>& logWeightDenominatorByDetectorPSet = conf.getParameterSetVector("logWeightDenominatorByDetector");
+
+      for( const auto& pset : logWeightDenominatorByDetectorPSet ) {
+	if(!pset.exists("detector")) {
+	  throw cms::Exception("logWeightDenominatorByDetectorPSet")
+	    << "logWeightDenominator : detector not specified";
+	}
```

Comment:
this ``` (!pset.exists("detector"))  ``` is redundant to 
``` const std::string& det = pset.getParameter<std::string>("detector"); ```
which will throw with a similar message if a detector pset is not available